### ISSUE TITLE
ImPlot Success

### ISSUE THREAD

**Issue by tomvdb** [ImGuiNET/ImGui.NET#435]
Hi,

I had the need to use ImPlot so went down the rabbit hole of getting it to work with ImGuiNet. The concept is fairly simple:

(1) Compile native cimplot dll files

to compile updated cimplot files I forked and modified the ImGui.NET-nativebuild repo. My version is available here (https://github.com/tomvdb/ImGui.NET-nativebuild)

(2) Generate new implot cs code using the codegenerator with updated definitions from cimplot

I then forked ImGui.NET and added the new dll files and cimplot definitions. The imgui controller required an update to set the implot context, but from there I could then load the implot demo window as well as create a seperate window with a barchart.

My forked version of ImGui.Net is at (https://github.com/tomvdb/ImGui.NET)

I have only tried native windows builds for implot for now. 

![image](https://github.com/ImGuiNET/ImGui.NET/assets/1420347/9b57ba4f-4acf-4c87-bdc0-684684705707)

Its probably all too messy for a PR, but that is where I'm at if its useful for someone

**Comment by gregkwaste** [comment]
Hi @tomvdb. I'm trying to follow the same procedure for the ImGuizmo. I keep getting memory corruption errors which are 99.9% related to a mismatch between the definitions and the binaries.

Can you please elaborate the process a little bit more?

Thanks!

**Comment by tomvdb** [comment]
Hi @gregkwaste,

As far as I can tell you steps looks good. That's the process I followed as well.

Two extra notes from my side:

(1) I had a crash as well on my first run of implot. I enabled mixed mode debugging in my project in Visual Studio so that it would jump into the C file and actually show me why it crashed. That helped me to realize that I had to add some extra CreateContext changes into the imguirenderer in the example. ie, with the mixed mode debugging you can get the underlying issue.

(2) As you noted in your PS, imgui is outdated, so you will definitely have to revert to an older cimgui to use that. Or update it with the changes required. 

Hope that helps :)

**Comment by gregkwaste** [comment]
Hi @tomvdb thanks for your reply!

I also got ImGuizmo.NET to work! 

I repeated the process again with 1.89.7, and I first tested just the imgui library to see if I can get it to work. And it worked pretty much directly once I selected the correct commit (based on the corresponding release).

Now the crash that I was getting with ImGuizmo was related to me not knowing how to exactly use the library. I tested some other functions of the dll and they worked so I could tell that the wrapper was indeed built correctly. It turned out that I was trying to begin an Imguizmo frame prior to setting a matching context for the library. Once I did that I had to find out how exactly to used it in order to render things on the screen.

It is working properly now! FINALLY!

Thanks for the suggestions!

**Comment by tomvdb** [comment]
Awesome :) Nicely done!

**Comment by guillaC** [comment]
It could be interesting to have a NuGet package ;)

**Comment by sonoro1234** [comment]
All submodules of LuaJIT-ImGui are compatible with last cimgui version

**Comment by gregkwaste** [comment]
> The reason it's not done up till now is because I couldn't figure out the release process of these new/other libraries. For example, currently ImGui.NET is 1.89.9. How do I get 1.89.9 version of cimguizmo?

Imguizmo hasn't received any substantial update for a while, so I guess that its code is still compatible, with the latest Imgui. For me, it worked for 1.89.7, so I guess 1.89.9 should also work.

**Comment by zaafar** [comment]
> For me, it worked for 1.89.7, so I guess 1.89.9 should also work.

Please note that imgui doesn't follow the same versioning system that c# (.net) follows, so there might be some changes in 1.89.10 (or 1.90) that will break the imguizmo or other libraries. Given how often some of those repo are maintained or tested with the latest version of imgui (e.g. https://github.com/Nelarius/imnodes/pull/181), it will severely reduce the speed at which I want to release imgui.net. The other option is that Sonoro1234 or I maintain each and every repo that exists (e.g. https://github.com/sonoro1234/imnodes) so that we can fix the compatibility issues ASAP. I do not want to do that.

Maybe we can release a new package (e.g. ImGUIAndOtherLibs.NET) that is only updated once every dependency (looking at the libs mentioned here https://github.com/ImGuiNET/ImGui.NET-nativebuild/pull/3) is updated.

Sorry for the rant :)

**Comment by guillaC** [comment]
> > For me, it worked for 1.89.7, so I guess 1.89.9 should also work.
> 
> Please note that imgui doesn't follow the same versioning system that c# (.net) follows, so there might be some changes in 1.89.10 (or 1.90) that will break the imguizmo or other libraries. Given how often some of those repo are maintained or tested with the latest version of imgui (e.g. [Nelarius/imnodes#181](https://github.com/Nelarius/imnodes/pull/181)), it will severely reduce the speed at which I want to release imgui.net. The other option is that Sonoro1234 or I maintain each and every repo that exists (e.g. https://github.com/sonoro1234/imnodes) so that we can fix the compatibility issues ASAP. I do not want to do that.
> 
> Maybe we can release a new package (e.g. ImGUIAndOtherLibs.NET) that is only updated once every dependency (looking at the libs mentioned here [ImGuiNET/ImGui.NET-nativebuild#3](https://github.com/ImGuiNET/ImGui.NET-nativebuild/pull/3)) is updated.
> 
> Sorry for the rant :)

Your suggestion to create a new package, such as 'ImGUIAndOtherLibs.NET,' appears to be an interesting option for addressing compatibility issues. No need to apologize, thank you for your work.

**Comment by sonoro1234** [comment]
> The other option is that Sonoro1234 or I maintain each and every repo that exists (e.g. https://github.com/sonoro1234/imnodes) so that we can fix the compatibility issues ASAP. I do not want to do that.

Yes, cimnodes is the only repo affected because imnodes is stuck on imgui 1.89.4.

**Comment by tingspain** [comment]
> > The other option is that Sonoro1234 or I maintain each and every repo that exists (e.g. https://github.com/sonoro1234/imnodes) so that we can fix the compatibility issues ASAP. I do not want to do that.
> 
> Yes, cimnodes is the only repo affected because imnodes is stuck on imgui 1.89.4. See: [Nelarius/imnodes#180](https://github.com/Nelarius/imnodes/pull/180)

How about the new [ImNodeFlow](https://github.com/Fattorino/ImNodeFlow/tree/master)?

**Comment by zaafar** [comment]
o wow, we have another one.
these things keep growing :)
